packages/open-api-gateway/README.md

```diff
@@ -553,6 +552,30 @@ export class SampleApi extends Api {
 }
 ```
 
+#### Sharing Integrations
+
+If you would like to use the same integration for every operation, you can use the `Operations.all` method from your generated client to save repeating yourself, for example:
+
+```ts
+import { Operations } from 'my-api-typescript-client';
+import { Authorizers, Integrations } from '@aws-prototyping-sdk/open-api-gateway';
+import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs';
+import { Construct } from 'constructs';
+import { Api } from './api';
+
+export class SampleApi extends Api {
+  constructor(scope: Construct, id: string) {
+    super(scope, id, {
+      defaultAuthorizer: Authorizers.iam(),
+      // Use the same integration for every operation.
+      integrations: Operations.all({
+        integration: Integrations.lambda(new NodejsFunction(scope, 'say-hello')),
+      }),
+    });
```

Comment:
Might be better to support a "default integration", which would allow to specify a uniq integration for specific routes while using the "default" (Operations.all) for the rest.

With current implementation, say I have 100 routes of which "99 share integration" and "1 has uniq integration", do I have to list all 100 because 1 of them does not use the shared?

Reply:
I thought about this a little but didn't want to remove the type-safety in TypeScript where it forces you to provide an integration for every operation 😄.

I figured you can still achieve this use case fairly easily with something like:

```ts
integrations: {
  ...Operations.all({ integration: myIntegration }),
  otherOperation: { integration: myOtherIntegration },
},
```

In python it's a little uglier (but I fully blame python for that 😉 ):

```python
integrations = Operations.all(
  OpenApiIntegration(
    integration=my_integration
  ),
)
integrations.other_operation = OpenApiIntegration(
  integration=my_other_integration
)

self.api = Api(self, 'Api',
  default_authorizer=Authorizers.iam(),
  integrations=integrations,
)
```

Reply:
That works pretty well actually, much cleaner than I expected. Mostly just wanted to know it was possible, and with your example it definitely is - great work!

Reply:
I created discussion https://github.com/aws/aws-prototyping-sdk/discussions/241 for the mustache to AST/JSII idea.